pkg/login/social/azuread_oauth.go

```diff
@@ -22,6 +22,10 @@ import (
 	"github.com/grafana/grafana/pkg/util"
 )
 
+var (
+	errAzureADMissingGroups = &Error{"either the user does not have any group membership or the groups claim is missing from the token."}
```

Comment:
When you say surface, is this error something you want to bubble up to the user trying to connect? What would be the error status associated with it?
```suggestion
    errAzureADMissingGroupsMsg = "either the user does not have any group membership or the groups claim is missing from the token."
	errAzureADMissingGroups = errutil.NewBase(errutil.StatusForbidden, errutil.WithPlublicMessage(errAzureADMissingGroupsMsg)).Errorf(errAzureADMissingGroupsMsg)
```

Reply:
Yes, this message will be surfaced. We followed the current guideline which is using an `&Error{}` and the social service converts this using the `fromSocialErr` to return `errutil.Unauthorized`, relevant parts from social.go:
https://github.com/grafana/grafana/blob/1141dd62ab440513ad843eadc65066e1f4b6e3e1/pkg/services/authn/clients/oauth.go#L119-L125